To add script3 to the `de-go-analysis` module  (4/5) (#240)
stjude-dnb-binfcore/sc-rna-seq-snap: analyses/de-go-analysis/03-go-clusterprofiler.Rmd

  dir.create(go_plots_dir_condition_value1)}

# Source functions
source(paste0(analysis_dir, "/util/function-go-clusterprofiler-markers.R"))
```

```{r echo=FALSE, warning=FALSE}
opts_chunk$set(fig.align='center',
               external=TRUE,
               echo=FALSE,
               warning=FALSE,
               fig.pos='H')
a4width <- 8.3
a4height <- 11.7
```

# Gene Ontology (GO) enrichment analysis using clusterProfiler

In this step, we conduct Gene Ontology (GO) enrichment analysis to identify biological processes associated with cell-type-specific expression changes (e.g., differential expression). Using the [clusterProfiler](https://bioconductor.org/packages/devel/bioc/html/clusterProfiler.html) package, we assess whether any GO terms are significantly enriched among these gene sets. This approach helps interpret the possible functional implications of differential expression across clusters, i.e., predicted cell type.

## Overall

```{r go-clusterProfiler-overall, fig.width = 15, fig.height = 8, fig.fullwidth = TRUE, echo=TRUE}
all.markers_overall <- read.csv(markers_file_overall, sep = "\t", header = TRUE)
top_n_value_overall <- all.markers_overall %>% 

>   group_by(cluster) 

Reviewer: Removing the top `n` filter looks good, but I still think it would make sense to filter the input genes for the GO analysis to genes which are being considered as significant (e.g. per the filters used for the volcano plotting, which are the `pCutoff` and `FCcutoff` defaults from the `EnhancedVolcano` function), otherwise this is using the full list of genes from the DE markers file.

```suggestion
top_n_value_overall <- all.markers_overall %>% 
  group_by(cluster) %>%
  dplyr::filter(p_val < 1e-05) %>%
  dplyr::filter(abs(avg_log2FC) > 1) 
```

Author: Added here: c1d0c939fcc3f72c4c06faa4494d958bdce130ed.

ego_results <- run_clusterwise_GO_enrichment_markers(closest_genes_all = top_n_value_overall, 
                                                     OrgDb_value = OrgDb_value, 
                                                     plots_dir = go_plots_dir_overall,

>                                                      universe_value = universe_value,

Reviewer: For the non-NULL `universe_value` option, I believe the idea is to pass the list of genes from the `seurat_obj` used for generating the DE results (before any gene filtering for FC, p-value, etc.). But based on the `names(geneList)`option defined in the `yaml` for PR #234, the list of gene names is not going to be passed to the function. `geneList` doesn't appear to be defined anywhere in this script, so I think `names(geneList)` will be empty.

Author: Let's address this to the PR that was updated.

                                                     show_number_pathways = show_number_pathways)
```

## Per `r condition_value1`

```{r go-clusterProfiler-condition-value1, fig.width = 15, fig.height = 8, fig.fullwidth = TRUE, echo=TRUE}
all.markers_condition_value1 <- read.csv(markers_file_condition_value1, sep = "\t", header = TRUE)
top_n_value_condition_value1 <- all.markers_condition_value1 %>% 

>   group_by(cluster) #%>% 

Reviewer: ```suggestion
top_n_value_condition_value1 <- all.markers_condition_value1 %>% 
  group_by(cluster) %>%
  dplyr::filter(p_val < 1e-05) %>%
  dplyr::filter(abs(avg_log2FC) > 1) 
```

Author: Added here: c1d0c939fcc3f72c4c06faa4494d958bdce130ed.

ego_results <- run_clusterwise_GO_enrichment_markers(closest_genes_all = top_n_value_condition_value1, 
                                                     OrgDb_value = OrgDb_value, 
                                                     plots_dir = go_plots_dir_condition_value1,
                                                     universe_value = universe_value,
                                                     show_number_pathways = show_number_pathways)
```

## Per `r condition_value2`

```{r go-clusterProfiler-condition-value2, fig.width = 15, fig.height = 8, fig.fullwidth = TRUE, echo=TRUE}
if (!is.null(condition_value2)) {
  
  go_plots_dir_condition_value2 <- file.path(go_plots_dir, "03-go-clusterprofiler-overall-condition2") 
  if (!dir.exists(go_plots_dir_condition_value2)) {
    dir.create(go_plots_dir_condition_value2)}
  
  all.markers_condition_value2 <- read.csv(markers_file_condition_value2, sep = "\t", header = TRUE)
  top_n_value_condition_value2 <- all.markers_condition_value2 %>% 

>     group_by(cluster) #%>% 

Reviewer: ```suggestion
  top_n_value_condition_value2 <- all.markers_condition_value2 %>% 
  group_by(cluster) %>%
  dplyr::filter(p_val < 1e-05) %>%
  dplyr::filter(abs(avg_log2FC) > 1) 
```

Author: Added here: c1d0c939fcc3f72c4c06faa4494d958bdce130ed.

  ego_results <- run_clusterwise_GO_enrichment_markers(closest_genes_all = top_n_value_condition_value2, 
                                                     OrgDb_value = OrgDb_value, 
                                                     plots_dir = go_plots_dir_condition_value2,
                                                     universe_value = universe_value,
                                                     show_number_pathways = show_number_pathways)
}
```

## Per `r condition_value3`

```{r go-clusterProfiler-condition-value3, fig.width = 15, fig.height = 8, fig.fullwidth = TRUE, echo=TRUE}
if (!is.null(condition_value3)) {
  
  go_plots_dir_condition_value3 <- file.path(go_plots_dir, "04-go-clusterprofiler-overall-condition3") 
  if (!dir.exists(go_plots_dir_condition_value3)) {
    dir.create(go_plots_dir_condition_value3)}
  
  all.markers_condition_value3 <- read.csv(markers_file_condition_value3, sep = "\t", header = TRUE)
  top_n_value_condition_value3 <- all.markers_condition_value3 %>% 

>     group_by(cluster)

Reviewer: ```suggestion
  top_n_value_condition_value3 <- all.markers_condition_value3 %>% 
  group_by(cluster) %>%
  dplyr::filter(p_val < 1e-05) %>%
  dplyr::filter(abs(avg_log2FC) > 1) 
```

Author: Added here: c1d0c939fcc3f72c4c06faa4494d958bdce130ed.

body { /* Normal */
  font-size: 16px;
  font-style: Arial, Helvetica, sans-serif;}

h1 {color: var(--DNB_BINF_Core_color);
    font-size: 28px;
    margin-top: 50px;}

h2 {color: var(--DNB_BINF_Core_color);
    font-size: 20px;}

.list-group-item.active, .list-group-item.active:focus, .list-group-item.active:hover {
  background-color: var(--DNB_BINF_Core_color);}
</style>

<a href="https://wiki.stjude.org/display/CAB">

</a>

\pagebreak

<div class="reporthead"><br/>
**PI: `r params$PI_NAME`**  
**Project: `r params$PROJECT_NAME`**  
Task: `r params$TASK_ID`  
Project Lead(s): `r params$PROJECT_LEAD_NAME`  
Department: `r params$DEPARTMENT`  

<br />  

DNB Bioinformatics Core Analysis Team: 
<br />  

>**Lead Analyst(s): `r params$LEAD_ANALYSTS`**  
>Group Lead: `r params$GROUP_LEAD`  
<br />
>**Contact E-mail:** `r params$CONTACT_EMAIL`  
>**DNB Bioinformatics Core Pipeline:** `r params$PIPELINE`  

Date started: `r params$START_DATE`  
Date completed:  `r params$COMPLETION_DATE`  
Report generated: `r format(Sys.time(), '%H:%M:%S %Z %m/%d/%Y')` \

Reviewed by: _____________________   Date: ____________ \
</div>
\pagebreak
  
# Information about this notebook

> After completing clustering and cell type annotation in the scRNA‑seq workflow, a common next step is to functionally interpret the resulting gene sets. Gene Ontology (GO) enrichment enables biological insight by identifying over‑represented biological processes, molecular functions, or cellular components among differentially expressed genes (DEGs).

Reviewer: ```suggestion
After completing clustering, cell type annotation, and per cell type differential expression (DE) analysis in the scRNA‑seq workflow, a common next step is to functionally interpret the resulting gene sets. Gene Ontology (GO) enrichment enables biological insight by identifying over‑represented biological processes, molecular functions, or cellular components among differentially expressed genes (DEGs).
```

markers_file_condition_value3 <- file.path(markers_dir_module, glue::glue("Res_{resolution_list}_Markers_all_condition_value3.tsv"))

# Create plots directory
go_plots_dir <- file.path(module_plots_dir, "03-go-clusterprofiler") 
if (!dir.exists(go_plots_dir)) {
  dir.create(go_plots_dir)}

# Create plots directory
go_plots_dir_overall <- file.path(go_plots_dir, "01-go-clusterprofiler-overall") 
if (!dir.exists(go_plots_dir_overall)) {
  dir.create(go_plots_dir_overall)}

# Create plots directory
go_plots_dir_condition_value1 <- file.path(go_plots_dir, "02-go-clusterprofiler-overall-condition1") 
if (!dir.exists(go_plots_dir_condition_value1)) {
  dir.create(go_plots_dir_condition_value1)}

# Source functions
source(paste0(analysis_dir, "/util/function-go-clusterprofiler-markers.R"))
```

```{r echo=FALSE, warning=FALSE}
opts_chunk$set(fig.align='center',
               external=TRUE,
               echo=FALSE,
               warning=FALSE,
               fig.pos='H')
a4width <- 8.3
a4height <- 11.7
```

# Gene Ontology (GO) enrichment analysis using clusterProfiler

> In this step, we conduct Gene Ontology (GO) enrichment analysis to identify biological processes associated with cluster-specific accessible regions (e.g., differential expression). Using the [clusterProfiler](https://bioconductor.org/packages/devel/bioc/html/clusterProfiler.html) package, we assess whether any GO terms are significantly enriched among these gene sets. This approach helps interpret the functional implications of differential expression across clusters, i.e., predicted cell type.

Reviewer: ```suggestion
In this step, we conduct Gene Ontology (GO) enrichment analysis to identify biological processes associated with cell-type-specific expression changes (e.g., differential expression). Using the [clusterProfiler](https://bioconductor.org/packages/devel/bioc/html/clusterProfiler.html) package, we assess whether any GO terms are significantly enriched among these gene sets. This approach helps interpret the functional implications of differential expression across clusters, i.e., predicted cell type.
```

> In this step, we conduct Gene Ontology (GO) enrichment analysis to identify biological processes associated with cluster-specific accessible regions (e.g., differential expression). Using the [clusterProfiler](https://bioconductor.org/packages/devel/bioc/html/clusterProfiler.html) package, we assess whether any GO terms are significantly enriched among these gene sets. This approach helps interpret the functional implications of differential expression across clusters, i.e., predicted cell type.

Author: See: b82014666b8637f3ad5fd268131a0b319af0237e.

> We will filter and keep `r n_value` markers for visualization purposes.

Author: I addressed this here: c0cb40eaea05cb91cf127750a7dd3a25b51d74c1. Please note the following:

- Added new variables about the universe parameter and the number of pathways to include in the plots in the GO function
- Fixed documentation where needed.
